### PR TITLE
vendor: docker and docker-cli v19.03.15

### PR DESCRIPTION
**Pull request by greut**
:clap:  congrats on the move to go modules!

How did you handle the files that aren't fully compatible with the latest x/sys/windows? https://github.com/moby/moby/commit/c3a0a3744636069f43197eb18245aaae89f568e5

### CODE REVIEW

**Comment by shoenig** [comment]
A lot of the work around upgrading and normalizing our dependencies was done by you, so thank you, @greut ! We've still got work to do, including eliminating the vendor directory, publishing versions for `api`, and converting `plugins` into an independent module.

TBH we didn't need to do anything special for docker (yet). I just set the version in `go.mod` as the equivelent psuedo version of the revision set in the old `vendor.json`. That brought in newer versions of a lot of transitive dependencies, but not in a way that caused a breakage, fortunately. I'm not sure what needs to be done to upgrade further, but these newest versions are evidently going to be a problem 

```
-       github.com/docker/cli v0.0.0-20200303215952-eb310fca4956
+       github.com/docker/cli v0.0.0-20200528204125-dd360c7c0de8
        github.com/docker/distribution v2.7.1+incompatible
-       github.com/docker/docker v17.12.0-ce-rc1.0.20200330121334-7f8b4b621b5d+incompatible
+       github.com/docker/docker v17.12.0-ce-rc1.0.20200531234253-77e06fda0c94+incompatible
```

```
$ GOOS=windows go install
# github.com/docker/docker/pkg/system
vendor/github.com/docker/docker/pkg/system/filesys_windows.go:111:24: cannot use uintptr(unsafe.Pointer(&sd[0])) (type uintptr) as type *"golang.org/x/sys/windows".SECURITY_DESCRIPTOR in assignment
```

So basically `golang.org/x/sys/windows` and `github.com/docker/docker` need to be in agreement version-wise, but I'm not sure what to do to actually resolve conflict against a built-in package 🙁

**Comment by greut** [comment]
@shoenig there was a _hack_ where some bits of the vendors were fetched from a different commit. https://github.com/hashicorp/nomad/pull/8042/commits/1b61b16c01e69cf8d6d06b50e82a88a6aa22125d#diff-bd290170e2912d3e8694db1a151066e5

If it works as is, keep it that way :)

**Comment by supernovae** [comment]
any chance we can see this feature soon?

**Comment by greut** [comment]
> any chance we can see this feature soon?

it should be tried again with docker v19.03.12

**Comment by greut** [comment]
This is the only difference with upstream Docker v19.03.13 that needed to be kept is the following change (picked from docker/master)

cf. https://github.com/moby/moby/commit/c3a0a3744636069f43197eb18245aaae89f568e5

```patch
diff --git a/vendor/github.com/docker/docker/pkg/system/filesys_windows.go b/vendor/github.com/docker/docker/pkg/system/filesys_windows.go
index c4d890f27..3049ff38a 100644
--- a/vendor/github.com/docker/docker/pkg/system/filesys_windows.go
+++ b/vendor/github.com/docker/docker/pkg/system/filesys_windows.go
@@ -11,6 +11,7 @@ import (
 	"time"
 	"unsafe"
 
-	winio "github.com/Microsoft/go-winio"
 	"golang.org/x/sys/windows"
 )
 
@@ -101,13 +102,13 @@ func mkdirall(path string, applyACL bool, sddl string) error {
 // and Local System.
 func mkdirWithACL(name string, sddl string) error {
 	sa := windows.SecurityAttributes{Length: 0}
-	sd, err := winio.SddlToSecurityDescriptor(sddl)
+	sd, err := windows.SecurityDescriptorFromString(sddl)
 	if err != nil {
 		return &os.PathError{Op: "mkdir", Path: name, Err: err}
 	}
 	sa.Length = uint32(unsafe.Sizeof(sa))
 	sa.InheritHandle = 1
-	sa.SecurityDescriptor = uintptr(unsafe.Pointer(&sd[0]))
+	sa.SecurityDescriptor = sd
 
 	namep, err := windows.UTF16PtrFromString(name)
 	if err != nil {
```

**Comment by shoenig** [comment]
i believe this got done in the 1.3 cycle / runc upgrades

```
➜ grep docker/docker go.mod
	github.com/docker/docker v20.10.12+incompatible
	github.com/docker/docker-credential-helpers v0.6.4 // indirect
```
